### DISCUSSION TITLE
Meteoroids Game

### DISCUSSION THREAD

**Discussion by jzeiber** [aduros/wasm4#57] [General]
I created a simple game in C.  I would have preferred to use C++, but the linker was complaining about missing symbols for global destructors and vtables and I couldn't get it to work.

Anyway, you can find the source and cart here.  Enjoy.
https://github.com/jzeiber/wasm4-meteoroids

I also created the bundle with the w4 tool, but it doesn't seem to work.  Chrome has this message in the console:
`meteoroids.html:25 Uncaught (in promise) CompileError: WebAssembly.instantiate(): expected magic word 00 61 73 6d, found 00 00 00 00 @+0`

**Comment by aduros** [comment]
This is awesome!! Nice use of rotation, and the controls are perfect! Would you be interested in hosting the cart on wasm4.org?

I'm looking into that bundle bug, thanks for reporting.

**Reply by aduros** [reply]
Just following up to say the bundle bug was fixed :nerd_face: 

If you feel like it, please open a PR to add your cart to wasm4.org. It would be really great to have such a high quality example!

**Reply by jzeiber** [reply]
Thanks.  The bundle is working now.  https://jzeiber.github.io/wasm4-meteoroids/.  I think it needs some more polish, particularly some better SFX, which I am not great at.

I have noticed on different computes and browsers different behavior of the cartridge.  On Chrome on one computer it runs at about 50% speed, so about 30 ticks per second.  On another computer with Chrome it runs fine at full speed.  I also noticed the sound wasn't played back correctly on a laptop in Firefox, it cuts off abruptly and the frequency is off, but on the same laptop with Waterfox it was working as expected.

I see that the diskr and diskw functions take a buffer and size, but not an offset.  Essentially if you want to read or write just the last byte you need a buffer of 1024 bytes, which normally wouldn't be an issue but with such limited memory it could be.  I suggest either adding an offset to these functions, or new versions of them that take an offset.
